base/genericmemory.jl

```diff
@@ -119,15 +119,15 @@ function unsafe_copyto!(dest::Memory, doffs, src::Memory, soffs, n)
     @inbounds if destp < srcp || destp > endp
         for i = 1:n
             if isassigned(src, soffs + i - 1)
-                dest[doffs + i - 1] = src[soffs + i - 1]
+                dest[doffs + i - 1] = src[soffs + i - 1]::T
```

Comment:
I think this is missing a conversion?

Reply:
Ah, good point - Unfortunately that will re-introduce the type instability.